### ISSUE TITLE
false positive, not a structure

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#2830]
original issue by @dpuerner

```abap
TYPES: BEGIN OF ty_pil,
         pid TYPE i,
         lbay TYPE i,
       END OF ty_pil.

DATA lt_pil TYPE TABLE OF ty_pil.

LOOP AT lt_pil INTO DATA(ls_pil) GROUP BY ( pid = ls_pil-pid ) ASCENDING REFERENCE INTO DATA(ls_pil2_ref).
  WRITE 'Hello World'.
ENDLOOP.
```

![image](https://user-images.githubusercontent.com/5888506/210781263-cd1c238d-8399-4184-b09b-7efc432982bc.png)
